admin/add_copyright.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env PYTHONPATH=./gcovr python3
+
+# -*- coding:utf-8 -*-
+
+#  ************************** Copyrights and license ***************************
+#
+# This file is part of gcovr 4.2, a parsing and reporting tool for gcov.
+# https://gcovr.com/en/stable
+#
+# _____________________________________________________________________________
+#
+# Copyright (c) 2021 the gcovr authors
+# Copyright (c) 2013 Sandia Corporation.
+# This software is distributed under the BSD License.
+# Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+# the U.S. Government retains certain rights in this software.
+# For more information, see the README.rst file.
+#
+# ****************************************************************************
+
+import os
+import re
+import subprocess
+
+import version
+
+REGEX_EMAIL = re.compile(r"\d+\+([^@]+)@users\.noreply\.github\.com")
+HEADER_END = (
+    " ****************************************************************************"
+)
+
+
+def getLicenseSection(filename, comment_char="#"):
+    date = subprocess.check_output(
+        ["git", "log", "-1", "--format=format:%ad", "--date=short", "--", filename],
+        universal_newlines=True,
+    )
+    year = date[:4]
+    yield comment_char + "  ************************** Copyrights and license ***************************"
+    yield comment_char
+    yield comment_char + " This file is part of gcovr {}, a parsing and reporting tool for gcov.".format(
```

Comment:
If we drop support for Python 3.5 (which is EOL since last year), we could use f-strings: `f"foo {variable} bar"`

Reply:
In the test it's already dropped but I think for the next release we shouldn't use this. After the release we can change every code places to format strings.